In [1]:
import pandas as pd
import numpy as np
import io
import os

# 1. Directly create the starter dataset inside the notebook
csv_data = """student_id,score,passed
1,85,Yes
2,42,No
3,90,Yes
4,65,Yes
5,50,No"""

# Save it locally into your data folder to satisfy the task directory structure
os.makedirs('data', exist_ok=True)
with open('data/starter_data.csv', 'w') as f:
    f.write(csv_data)

# 2. Ingest the dataset into a DataFrame
df = pd.read_csv(io.StringIO(csv_data))

# 3. Print the required outputs: shape, columns, and head
print("--- 1. DATASET SHAPE ---")
print(df.shape)

print("\n--- 2. COLUMNS ---")
print(df.columns.tolist())

print("\n--- 3. HEAD (FIRST ROWS) ---")
display(df.head())

print("\n--- 4. SMOKE TEST AGGREGATES ---")
print(f"Total Row Count: {len(df)}")
print(f"Column Mean (Average Score): {df['score'].mean()}")

--- 1. DATASET SHAPE ---
(5, 3)

--- 2. COLUMNS ---
['student_id', 'score', 'passed']

--- 3. HEAD (FIRST ROWS) ---


,student_id,score,passed
0,1,85,Yes
1,2,42,No
2,3,90,Yes
3,4,65,Yes
4,5,50,No



--- 4. SMOKE TEST AGGREGATES ---
Total Row Count: 5
Column Mean (Average Score): 66.4


### Data Dictionary (First Draft)

| Column Name | Data Type | Description | Notes / Observations |
|---|---|---|---|
| `student_id` | int64 | Unique identification number for each student | Serves as the primary key; no missing values. |
| `score` | int64 | Test mark out of 100 | Continuous numeric value; mean score is 66.4. |
| `passed` | object | Completion status indication (`Yes` or `No`) | Categorical text value tracking evaluation results. |
